optuna/distributions.py

```diff
@@ -477,14 +477,52 @@ def json_to_distribution(json_str: str) -> BaseDistribution:
 
     json_dict = json.loads(json_str)
 
-    if json_dict["name"] == CategoricalDistribution.__name__:
-        json_dict["attributes"]["choices"] = tuple(json_dict["attributes"]["choices"])
-
-    for cls in DISTRIBUTION_CLASSES:
-        if json_dict["name"] == cls.__name__:
-            return cls(**json_dict["attributes"])
-
-    raise ValueError("Unknown distribution class: {}".format(json_dict["name"]))
+    if "name" in json_dict:
+        if json_dict["name"] == CategoricalDistribution.__name__:
+            json_dict["attributes"]["choices"] = tuple(json_dict["attributes"]["choices"])
+
+        for cls in DISTRIBUTION_CLASSES:
+            if json_dict["name"] == cls.__name__:
+                return cls(**json_dict["attributes"])
+
+        raise ValueError("Unknown distribution class: {}".format(json_dict["name"]))
+
+    else:
+        # Deserialize a distribution from an abbreviated format.
+        if json_dict["type"] == "categorical":
+            return CategoricalDistribution(json_dict["choices"])
+        elif json_dict["type"] in ("float", "int"):
+            low = json_dict["low"]
+            high = json_dict["high"]
+            step = json_dict.get("step")
+            log = json_dict.get("log")
+
+            if json_dict["type"] == "float":
+                if log:
+                    if step:
+                        raise ValueError(
+                            "The parameter `step` is not supported when `log` is true."
+                        )
+                    else:
+                        return LogUniformDistribution(low, high)
+                else:
+                    if step:
+                        return DiscreteUniformDistribution(low, high, step)
+                    else:
+                        return UniformDistribution(low, high)
+            else:
+                if log:
+                    if step:
+                        return IntLogUniformDistribution(low, high, step)
+                    else:
+                        return IntLogUniformDistribution(low, high)
+                else:
+                    if step:
+                        return IntUniformDistribution(low, high, step)
+                    else:
+                        return IntUniformDistribution(low, high)
```

Comment:
I found that `step=0` is implicitly changed to `step=1`. 

```python
>>> optuna.distributions.json_to_distribution('{"type": "int", "low": 1, "high": 10, "step": 0}')
{'type': 'int', 'low': 1, 'high': 10, 'step': 0}
IntUniformDistribution(high=10, low=1, step=1)
```

If I tried to create `IntUniformDistribution` with `step=0`, the following error occurred.
How about raising `ValueError` if `step=0`?

```python
>>> optuna.distributions.IntUniformDistribution(low=1, high=10, step=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/yanase/pfn/code/optuna-review/optuna/distributions.py", line 268, in __init__
    raise ValueError(
ValueError: The `step` value must be non-zero positive value, but step=0.
```

I found the same issue in `IntUniformDistribution` and `UniformDistribution`.

`IntLogUniformDistribution`:
```python
>>> optuna.distributions.json_to_distribution('{"type": "int", "low": 1, "high": 10, "step": 0, "log": true}')
{'type': 'int', 'low': 1, 'high': 10, 'step': 0, 'log': True}
IntLogUniformDistribution(high=10, low=1, step=1)
```

`UniformDistribution`:
```python
>>> optuna.distributions.json_to_distribution('{"type": "float", "low": 1, "high": 10, "step": 0}')
{'type': 'float', 'low': 1, 'high': 10, 'step': 0}
UniformDistribution(high=10.0, low=1.0)
```